src/components/categories/jokes/Jokes.vue

```diff
@@ -65,34 +68,46 @@
                 this.sendMessage("user", text)
             },
             showJoke(category) {
-                if (category != null) {
-                    this.sendMessageFromUser(this.$t('jokes.user.categoryChosen') + category);
-                    jokesService.getJokeForCategory(category).then(
+                this.error = '';
+                if (category == null) {
+                    this.sendMessageFromUser(this.$t('jokes.user.randomCategoryChosen'));
+                    jokesService.getRandomJoke().then(
                         (data) => {
-                            this.shownJoke = data;
-                            this.sendMessageFromBot(this.shownJoke.joke);
-                            this.sendMessageFromBot(this.$t('jokes.bot.nextStep'))
+                            if (data.errors) {
```

Comment:
Tutaj użyjmy gotowego mechanizmu, który mamy. A mianowicie komponentu Dialog i wydarzenia  'showDanger'. Podobnie we wszystkich przypadkach, gdzie pobieramy dane